dadi/lib/handlers/image.js

```diff
@@ -966,6 +978,36 @@ ImageHandler.prototype.process = function (imageBuffer, options, imageInfo) {
   })
 }
 
+/**
+ * Transcodes an input buffer to a GIF, ensures colours are appropriate
+ * for GIF encoding via a "quantize" method.
+ *
+ * Saves the encoded GIF to a temporary file and removes to before returning
+ * the buffer.
+ *
+ * @param {Buffer} buffer - a Buffer extracted from the main image
+ * processor after applying image manipulations
+ * @returns {Buffer} a GIF encoded buffer
+ */
+ImageHandler.prototype.processGif = function (buffer) {
```

Comment:
It seems to me like you don't need to create a wrapping Promise, since both `Jimp.read` and `GifUtil.write` return one already. Could you do:

```js
return Jimp.read(buffer).then(image => {
  // (...)

  return GifUtil.write(tmpGifFile, [frame]).then(gif => {
    // (...)
```

Reply:
👍